### ISSUE TITLE
SPWebAppAuthentication: Basic authentication

### ISSUE THREAD

**Issue by egor-yudkin** [PowerShell/SharePointDsc#1010]
#### Details of the scenario you tried and the problem that is occurring
I'd like to configure Basic authentication along with NTLM for web application using SPWebAppAuthentication DSC resource. But it seems MSFT_SPWebAppAuthenticationMode class doesn't have "Basic" in AuthenticationMethod parameter, so I assume it's not a supported authentication method in SharePointDSC/MSFT_SPWebAppAuthentication.

#### Verbose logs showing the problem
N/A

#### Suggested solution to the issue
Implement Basic authentication capability in MSFT_SPWebAppAuthentication DSC resource

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
I'd like to have something like this fake code possible
```powershell
            SPWebAppAuthentication ContosoAuthentication
            {
                WebAppUrl = "http://sharepoint.contoso.com"
                Default = @(
                    MSFT_SPWebAppAuthenticationMode{
                        AuthenticationMethod = "NTLM"
                    }
                    MSFT_SPWebAppAuthenticationMode{
                        AuthenticationMethod = "Basic"
                    }
                )
            }
```

#### The operating system the target node is running
OsName               : Microsoft Windows Server 2012 R2 Standard
OsOperatingSystemSKU : StandardServerEdition
OsArchitecture       : 64-bit
WindowsBuildLabEx    : 9600.19202.amd64fre.winblue_ltsb.181110-0600
OsLanguage           : en-US
OsMuiLanguages       : {en-US}

#### Version of SharePoint that is used (e.g. SharePoint 2016)
SharePoint 2016

#### Version and build of PowerShell the target node is running
PSVersion                      5.1.14409.1018                                  
PSEdition                      Desktop                                         
BuildVersion                   10.0.14409.1018                                 

#### Version of the DSC module that was used ('dev' if using current dev branch)
2.0.0.0

**Comment by ykuijs** [comment]
Hi @egor-yudkin You are correct, SharePointDsc does not yet support Basic authentication. I have just been investigating the implications of adding support for Basic auth: Right now we have the possibility to specify either NTLM or Kerberos. However Basic auth is a property that can be specified seperately. In the current setup this is not possible, you would only be able to specify NTLM/Kerberos **with** Basic auth.

In order to implement this, we need to update the resource to be able to specify Windows Authentication and then specify to use NTLM or Kerberos and to use Basic or not. This does however mean a breaking change. For example:

`
MSFT_SPWebAppAuthenticationMode{
    AuthenticationMethod = "Windows Authentication"
    WindowsAuthentication = "<NTLM/Kerberos>"
    BasicAuthentication = $true/$false
}
`

As a workaround we can implement Basic auth to only work with NTLM/Kerberos and extend this to be able to only specify Basic Auth with the v4.0 release. Would this work for you?

**Comment by egor-yudkin** [comment]
@ykuijs for now I'm just using a workaround with a regular PowerShell script to set up authentication, so in my case there is no rush. I can definitely wait for release 4.0 to see this feature implemented.

**Comment by ykuijs** [comment]
Implemented via PR #1133 in the v4.0-development branch. Will be included in v4.0.
